### PR TITLE
refine the KMS admin API

### PR DESCRIPTION
**Pull request by aead**
## Description
This commit removes the `Update` functionality
from the admin API. While this is technically
a breaking change I think this will not cause
any harm because:
 - The KMS admin API is not complete, yet.
   At the moment only the status can be fetched.
 - The `mc` integration hasn't been merged yet.
   So no `mc` client could have used this API
   in the past.

The `Update`/`Rewrap` status is not useful anymore.
It provided a way to migrate from one master key version
to another. However, KES does not support the concept of
key versions. Instead, key migration should be implemented
as migration from one master key to another.

Basically, the `Update` functionality has been implemented just
for Vault.

## Motivation and Context
KMS admin API minio/mc#2882

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#1c9a4c85-0575-4a0f-ae7c-37b0c33049f1) |
| mint-large-bucket.sh | :x: [more...](#a47a8e85-aa07-4445-a742-f9b5b281bf60) |
| mint-gateway-azure.sh | :x: [more...](#43e962fe-fdc1-489d-9dfe-3ec121fd0a76) |
<a name="1c9a4c85-0575-4a0f-ae7c-37b0c33049f1"></a>
### 8943-d9fe385/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:32421
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a89e92cc279a:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 13 seconds
{
  "name": "awscli",
  "duration": 10808,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="a47a8e85-aa07-4445-a742-f9b5b281bf60"></a>
### 8943-d9fe385/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31498
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ed37b99b1ff0:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 26 seconds
(2/13) Running aws-sdk-java tests ... done in 2 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 49 seconds
(5/13) Running minio-go tests ... FAILED in 15 minutes and 11 seconds
{
  "args": {},
  "duration": 96259,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="43e962fe-fdc1-489d-9dfe-3ec121fd0a76"></a>
### 8943-d9fe385/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30136
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c4cc91076b66:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 4 seconds
{
  "name": "awscli",
  "duration": 2013,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by Praveenrajmani — APPROVED** [review]
Makes sense. LGTM
